src/state/v1.rs

```diff
@@ -758,20 +760,107 @@ impl LoaderPolicySpec {
 /// Policy for zone key management.
 #[derive(Clone, Debug, Deserialize, Serialize)]
 #[serde(rename_all = "kebab-case", deny_unknown_fields)]
-pub struct KeyManagerPolicySpec {}
+pub struct KeyManagerPolicySpec {
+    /// Whether to use a CSK (if true) or a KSK and a ZSK.
+    use_csk: bool,
+
+    /// Algorithm and other parameters for key generation.
+    algorithm: KeyParameters,
+
+    /// Validity of KSKs.
+    ksk_validity: Option<u64>,
+    /// Validity of ZSKs.
+    zsk_validity: Option<u64>,
+    /// Validity of CSKs.
+    csk_validity: Option<u64>,
+
+    /// Configuration variable for automatic KSK rolls.
+    auto_ksk: AutoConfig,
+    /// Configuration variable for automatic ZSK rolls.
+    auto_zsk: AutoConfig,
+    /// Configuration variable for automatic CSK rolls.
+    auto_csk: AutoConfig,
+    /// Configuration variable for automatic algorithm rolls.
+    auto_algorithm: AutoConfig,
+
+    /// DNSKEY signature inception offset (positive values are subtracted
+    ///from the current time).
+    dnskey_inception_offset: u64,
+
+    /// DNSKEY signature lifetime
+    dnskey_signature_lifetime: u64,
+
+    /// The required remaining signature lifetime.
+    dnskey_remain_time: u64,
+
+    /// CDS/CDNSKEY signature inception offset
+    cds_inception_offset: u64,
+
+    /// CDS/CDNSKEY signature lifetime
+    cds_signature_lifetime: u64,
+
+    /// The required remaining signature lifetime.
+    cds_remain_time: u64,
+
+    /// The DS hash algorithm.
+    ds_algorithm: DsAlgorithm,
+
+    /// The TTL to use when creating DNSKEY/CDS/CDNSKEY records.
+    default_ttl: Ttl,
+
+    /// Automatically remove keys that are no long in use.
+    autoremove: bool,
```

Comment:
Is that immediate, or after a predictable or unpredictable delay, or ...? Knot has `delete-delay: TIME` (it is a minimum it seems, and is 0 by default) and BIND9 has `purge-keys <duration>` (defaults to 90 days, is added to the "pre-publication interval"). Should we enhance the comment here to indicate when removal will happen? I don't mean to suggest that we should add support for a delay period btw, but should we track it as a future TODO?

Reply:
I add a comment where the default is set to true.